lib/terraforming/resource/route53_record.rb

```diff
@@ -66,18 +73,27 @@ def record_sets_of(hosted_zone)
       end
 
       def records
-        hosted_zones.map do |hosted_zone|
+        to_return = hosted_zones.map do |hosted_zone|
           record_sets_of(hosted_zone).map { |record| { record: record, zone_id: zone_id_of(hosted_zone) } }
         end.flatten
+        count = {}
+        dups = to_return.group_by { |record| module_name_of(record[:record], nil) }.select { |_, v| v.size > 1 }.map(&:first)
+        to_return.each do |r|
+          module_name = module_name_of(r[:record], nil)
+          next unless dups.include?(module_name)
+          count[module_name] = count[module_name] ? count[module_name] + 1 : 0
+          r[:counter] = count[module_name]
+        end
+        to_return
       end
 
       # TODO(dtan4): change method name...
       def name_of(dns_name)
         dns_name.gsub(/\.\z/, "")
       end
 
-      def module_name_of(record)
-        normalize_module_name(name_of(record.name) + "-" + record.type)
+      def module_name_of(record, counter)
```

Comment:
__[method_definition_validator]__
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/iam_user.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/iam_user.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/eip.rb#L35](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/eip.rb#L35)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/launch_configuration.rb#L48](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/launch_configuration.rb#L48)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/internet_gateway.rb#L31](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/internet_gateway.rb#L31)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/alb.rb#L40](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/alb.rb#L40)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/elb.rb#L51](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/elb.rb#L51)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/network_acl.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/network_acl.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/kms_key.rb#L24](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/kms_key.rb#L24)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/route53_zone.rb#L37](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/route53_zone.rb#L37)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/elasti_cache_cluster.rb#L45](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/elasti_cache_cluster.rb#L45)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/elasti_cache_subnet_group.rb#L29](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/elasti_cache_subnet_group.rb#L29)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/db_security_group.rb#L30](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/db_security_group.rb#L30)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/iam_policy_attachment.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/iam_policy_attachment.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/db_parameter_group.rb#L31](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/db_parameter_group.rb#L31)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/auto_scaling_group.rb#L52](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/auto_scaling_group.rb#L52)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/iam_policy.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/iam_policy.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/cloud_watch_alarm.rb#L24](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/cloud_watch_alarm.rb#L24)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/nat_gateway.rb#L34](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/nat_gateway.rb#L34)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/vpc.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/vpc.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/iam_group_membership.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/iam_group_membership.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/network_interface.rb#L33](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/network_interface.rb#L33)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/iam_instance_profile.rb#L31](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/iam_instance_profile.rb#L31)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/efs_file_system.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/efs_file_system.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/db_subnet_group.rb#L29](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/db_subnet_group.rb#L29)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/s3.rb#L25](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/s3.rb#L25)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/security_group.rb#L36](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/security_group.rb#L36)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/sqs.rb#L25](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/sqs.rb#L25)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/sqs.rb#L36](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/sqs.rb#L36)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/redshift.rb#L44](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/redshift.rb#L44)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/iam_role.rb#L33](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/iam_role.rb#L33)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/ec2.rb#L53](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/ec2.rb#L53)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/subnet.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/subnet.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/kms_alias.rb#L24](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/kms_alias.rb#L24)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/vpn_gateway.rb#L32](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/vpn_gateway.rb#L32)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/rds.rb#L51](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/rds.rb#L51)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/iam_group.rb#L31](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/iam_group.rb#L31)
* Not enough arguments. Did you forget the following arguments? [counter] at [lib/terraforming/resource/route_table.rb#L33](/dtan4/terraforming/blob/9e9a1ffcfdc431d947655e4fa39e121cc886873d/lib/terraforming/resource/route_table.rb#L33)

<!-- Comment by [SideCI](https://sideci.com) -->